assets/src/edit-story/elements/text/display.js

```diff
@@ -33,19 +33,73 @@ import {
   elementWithTextParagraphStyle,
 } from '../shared';
 import StoryPropTypes from '../../types';
+import { BACKGROUND_TEXT_MODE } from '../../constants';
 import { useTransformHandler } from '../../components/transform';
-import { draftMarkupToContent, generateParagraphTextStyle } from './util';
-
-const Element = styled.p`
-	${elementFillContent}
-	${elementWithFont}
-	${elementWithBackgroundColor}
-	${elementWithFontColor}
-	${elementWithTextParagraphStyle}
+import {
+  draftMarkupToContent,
+  getHighlightLineheight,
+  generateParagraphTextStyle,
+} from './util';
+
+const HighlightElement = styled.p`
+  ${elementFillContent}
+  ${elementWithFont}
+  ${elementWithFontColor}
+  ${elementWithTextParagraphStyle}
+  line-height: ${({ lineHeight, verticalPadding }) =>
+    getHighlightLineheight(lineHeight, verticalPadding)};
+  margin: 0;
+  padding: 0;
+`;
+
+const HIGHLIGHT_MARGIN_BUFFER = 10;
+
+const MarginedElement = styled.span`
+  display: inline-block;
```

Comment:
Yes, it's necessary here. It can't be either `inline` or `block`, as it has to mediate the white space around the element correctly - there's some browser quirks around padding calculation for `box-decoration-break`.